storage/file.go

```diff
@@ -81,9 +192,94 @@ func (p ListSharesParameters) getParameters() url.Values {
 	return out
 }
 
-// pathForFileShare returns the URL path segment for a File Share resource
-func pathForFileShare(name string) string {
-	return fmt.Sprintf("/%s", name)
+func (p ListDirsAndFilesParameters) getParameters() url.Values {
+	out := url.Values{}
+
+	if p.Marker != "" {
+		out.Set("marker", p.Marker)
+	}
+	if p.MaxResults != 0 {
+		out.Set("maxresults", fmt.Sprintf("%v", p.MaxResults))
+	}
+	if p.Timeout != 0 {
+		out.Set("timeout", fmt.Sprintf("%v", p.Timeout))
+	}
+
+	return out
+}
+
+func (fr FileRange) String() string {
+	return fmt.Sprintf("bytes=%d-%d", fr.Start, fr.End)
+}
+
+// ToPathSegment returns the URL path segment for the specified values
+func ToPathSegment(parts ...string) string {
+	join := strings.Join(parts, "/")
+	if join[0] != '/' {
+		join = fmt.Sprintf("/%s", join)
+	}
+	return join
+}
+
+// returns url.Values for the specified types
+func getURLInitValues(comp compType, res resourceType) url.Values {
+	values := url.Values{}
+	if comp != compNone {
+		values.Set("comp", comp.String())
+	}
+	if res != resourceFile {
+		values.Set("restype", res.String())
+	}
+	return values
+}
+
+// ListDirsAndFiles returns a list of files or directories under the specified share or
+// directory.  It also contains a pagination token and other response details.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dn166980.aspx
+func (f FileServiceClient) ListDirsAndFiles(path string, params ListDirsAndFilesParameters) (DirsAndFilesListResponse, error) {
+	q := mergeParams(params.getParameters(), getURLInitValues(compList, resourceDirectory))
+
+	var out DirsAndFilesListResponse
+	resp, err := f.listContent(path, q, nil)
+	if err != nil {
+		return out, err
+	}
+
```

Comment:
I added the response code check to the listContent method so we only have to do it in one place (nice catch).